outboundSender.go

```diff
@@ -368,6 +387,28 @@ func (obs *CaduceusOutboundSender) Queue(msg *wrp.Message) {
 		return
 	}
 
+	//check the partnerIDs
+	if len(obs.listener.PartnerIDs) == 0 {
+		switch obs.noPIDAction {
```

Comment:
Consider making `noPIDAction` an enumerated type, and move the `switch/case` into a method of that type.  Here's one approach:

```
type NoPIDAction string

const (
    Custom NoPIDAction = "custom"
    DoNotSend NoPIDAction = "donotsend"
)

func (npa NoPIDAction) Execute(l Listener, pids []string) {
    switch {
        case Custom:
            l.PartnerIDs = pids
        case DoNotSend:
            // you could omit this "do nothing" case and just have a default that does nothing
        default:
            // can panic, return an error, do nothing, etc
    }
}
```

The advantage to this is that all of the logic for deciding what to do for a particular action is encapsulated in one place.  That reduces code noise and footprint of calling code, and it makes things easier to unit test.

Reply:
after today's conversation and some clarification from Kristina, we're going to get rid of this switch and simplify to just 2 use cases.